bin/pre-commit

```diff
@@ -5,50 +5,64 @@ PATH="$PATH:/usr/local/bin"
 printf "\nBy contributing to this project, you license the materials you contribute under the GNU General Public License v2 (or later). All materials must have GPLv2 compatible licenses — see CONTRIBUTING.md for details.\n\n"
 
 files=$(git diff --cached --name-only --diff-filter=ACM | grep ".jsx*$")
-if [ "$files" = "" ]; then
-    exit 0
-fi
-
 pass=true
 
-# i18nlinting.  Just have i18nlint issue warnings for now without failing
-printf "\nValidating translatable strings:\n"
-for file in ${files}; do
-    result=$(./bin/i18nlint --color ${file})
-    if [ $? -ne 0 ]; then
-        printf "\033[31mi18nlint Failed: \033[0m%s\n" "$result"
-        printf "\n -----\n"
-        pass=false
-    else
-        printf "\033[32mi18nlint Passed: %s\033[0m\n" "${file}"
-    fi
-done
-
-printf "\nValidating .jsx and .js:\n"
-
-for file in ${files}; do
-    ./node_modules/.bin/eslint ${file}
-    if [ $? -ne 0 ]; then
-        printf "\033[31meslint Failed: %s\033[0m\n" "${file}"
-        pass=false
-    else
-        printf "\033[32meslint Passed: %s\033[0m\n" "${file}"
-    fi
-done
-
-printf "\neslint validation complete\n"
-
-for file in ${files}; do
-    ./bin/gridiconFormatChecker ${file}
-        if [ $? -ne 0 ]; then
-        printf "\033[31mGridicon Format Check Failed: \033[0m%s\n" "${file}"
-        pass=false
-    fi
-done
+if [ "$files" != "" ]; then
+	# i18nlinting. Just have i18nlint issue warnings for now without failing
+	printf "\nValidating translatable strings:\n"
+	for file in ${files}; do
+		result=$(./bin/i18nlint --color ${file})
+		if [ $? -ne 0 ]; then
+			printf "\033[31mi18nlint Failed: \033[0m%s\n" "$result"
+			printf "\n -----\n"
+			pass=false
+		else
+			printf "\033[32mi18nlint Passed: %s\033[0m\n" "${file}"
+		fi
+	done
+
+	printf "\nValidating .jsx and .js:\n"
+
+	for file in ${files}; do
+		./node_modules/.bin/eslint ${file}
+		if [ $? -ne 0 ]; then
+			printf "\033[31meslint Failed: %s\033[0m\n" "${file}"
+			pass=false
+		else
+			printf "\033[32meslint Passed: %s\033[0m\n" "${file}"
+		fi
+	done
 
+	printf "\neslint validation complete\n"
+
+	for file in ${files}; do
+		./bin/gridiconFormatChecker ${file}
+		if [ $? -ne 0 ]; then
+			printf "\033[31mGridicon Format Check Failed: \033[0m%s\n" "${file}"
+			pass=false
+		fi
+	done
+fi
+
+packages=$(find client -type f -print | grep 'package.json')
```

Comment:
This will find folders where files have changed but those may be folders deeper in folder structure than where the package.json file is located.  We would need to step up folders to see if we find a package.json file.